features/vat-replacement/run.js

```diff
@@ -0,0 +1,239 @@
+                console.log('Update finished')
+            } else {
+                console.log('Current tax rate would be replaced as below : ')
+                console.log(JSON.stringify(updateJsonObj))
+            }
+        }
+
+    }
+}
+
+async function printPreviewModeWarning() {
+    console.log('********************************************************************')
+    console.log('The script has been run in preview mode. To update the tax rates, ' + '\n' +
+        'you can input argument "-update" in CLI to run this script.')
+    console.log('Please make sure to resolve all tax categories and rates warnings ' + '\n' +
+        'before running the script in update mode.')
+    console.log('********************************************************************')
+}
+
+async function initConfigOptions() {
+    const args = process.argv.slice(2);
+    const configOptions = {
+        ctp : {
+            projectKey : process.env.PROJECT_KEY,
+            clientId : process.env.CLIENT_ID,
+            clientSecret : process.env.CLIENT_SECRET
+        },
+        dryRun: true
+    }
+
+    // Check environment variables
+    if (!configOptions.ctp.projectKey || !configOptions.ctp.clientId || !configOptions.ctp.clientSecret) {
+        console.error('Please set project key, client ID and client secret in environment variables')
+        return
+    }
+
+    // Check update mode / preview mode
+    if (args.length>0 && args[0] === '-update') {
+        configOptions.dryRun = false
+    }
+    return configOptions
+}
+(async function main() {
+    const configOptions = initConfigOptions()
+    const ctpClient = ctpUtils.setUpClient(configOptions)
+
+    // Retrieve tax category list in current project
+    let taxCategories = await getTaxCategories(ctpClient)
+
+    // Initialize tax rate id <-> tax category mapping
+    const taxRateIdToTaxCategoryMap = await buildTaxRateIdToTaxCategoryMap(taxCategories)
+
+    // Validate the existing tax categories and returned valid german tax rates in project settings
+    const validGermanTaxRateList = await getGermanValidTaxRateList(taxCategories,
+        taxRateIdToTaxCategoryMap)
+
+    // Printout Json format TaxRateDraft and update it if it is not dry run.
+    await processTaxRate({
+        ctpClient: ctpClient,
+        validGermanTaxRateList: validGermanTaxRateList,
+        taxRateIdToTaxCategoryMap: taxRateIdToTaxCategoryMap,
+        taxRateType: vatConstant.TAX_RATE_TYPE_STANDARD,
+        isDryRun: configOptions.dryRun
+    })
+    await processTaxRate({
+        ctpClient: ctpClient,
+        validGermanTaxRateList: validGermanTaxRateList,
+        taxRateIdToTaxCategoryMap: taxRateIdToTaxCategoryMap,
+        taxRateType: vatConstant.TAX_RATE_TYPE_REDUCED,
+        isDryRun: configOptions.dryRun
+    })
+
+    if (configOptions.dryRun) {
+        await printPreviewModeWarning()
```

Comment:
You don't need await/async in this method.

Reply:
https://github.com/commercetools/commercetools-cli-scripts/pull/1/commits/5be9b0d46bfcbb447868073d9bb7167d0887abc1